doc/radosgw/qat-accel.rst

```diff
@@ -101,55 +112,132 @@ Configuration
 
-   The following CMake options have to be configured to trigger QAT based compression
-   when building Ceph:
+   For the out-of-tree QAT driver package, before starting ensure that both QAT
+   driver and `QATzip`_  have been installed. Besides ``ICP_ROOT``, remember to
+   set the environment variable ``QZ_ROOT`` to the root directory of your QATzip
+   source tree.
+
+   The following CMake options have to be configured to trigger QAT-based
+   compression when building Ceph:
   
    .. prompt:: bash $
 
-      ./do_cmake.sh -DWITH_QAT=ON -DWITH_QATZIP=ON
+      ./do_cmake.sh -DWITH_QATDRV=ON -DWITH_QATZIP=ON -DWITH_SYSTEM_QATZIP=ON -DWITH_QATLIB=OFF
 
-   Then, set an environment variable to clarify the section name of User Process Instance
-   Section in QAT configuration files, e.g.:
+   Set an environment variable to clarify the section name of User Process
+   Instance Section in the QAT configuration files. For example: 
   
    .. prompt:: bash $
 
       export QAT_SECTION_NAME=CEPH
 
-   Next, edit the Ceph configuration file to enable QAT support for compression::
+   **For in-tree QAT**
+
+   For in-tree QAT, make sure that your system meets the `QATlib System
+   Requirements`_.  QATlib can be installed from pre-built packages or from
+   source code.  See `QATlib Installation`_ . After QATlib is installed, you
+   can run ``cpa_sample_code`` to check if the QAT environment is OK.
+
+   If you are using QATlib source code, Ceph cmake build has the compilation
+   options of qatlib and qatzip enabled by default. Our normal compilation
+   already includes QAT-compressor-related code.
+
+   .. prompt:: bash $
+
+      ./do_cmake.sh
+
+   If you are using pre-built packages installed on the system, the following
+   CMake options have to be configured when building Ceph:
+
+   .. prompt:: bash $
+
+      ./do_cmake.sh -DWITH_SYSTEM_QATLIB=ON -DWITH_SYSTEM_QATZIP=ON
+
+
+   **For both out-of-tree QAT and in-tree QAT**
+
+   Edit the Ceph configuration file (usually ``ceph.conf``) to enable QAT
+   support for *zlib* compression::
 
       qat compressor enabled=true
 
+   Set the RGW compression method:
+
+   .. prompt:: bash $
+
+      # for storage class(STANDARD)
+      radosgw-admin zone placement modify --rgw-zone=default --placement-id=default-placement --compression=zlib
+      # or create a new storage class(COLD) and define data pool(default.rgw.cold.data)
+      radosgw-admin zonegroup placement add --rgw-zonegroup default --placement-id default-placement --storage-class COLD
+      radosgw-admin zone placement add --rgw-zone default --placement-id default-placement --storage-class COLD --compression zlib --data-pool default.rgw.cold.data
+
+CONFIG REFERENCE
+================
+The following QAT-related settings can be added to the Ceph configuration file
+(usually `ceph.conf`) under the ``[client.rgw.{instance-name}]`` section.
+
+.. confval:: qat_compressor_session_max_number
+.. confval:: qat_compressor_data_fmt
```

Comment:
this option was removed from https://github.com/ceph/ceph/pull/58046 during review, it can be removed here

Reply:
``qat_compressor_data_fmt``?

Reply:
Answer to my own question, based on review of https://github.com/ceph/ceph/pull/58046/files: yes.